pennylane/labs/resource_estimation/ops/qubit/non_parametric_ops.py

```diff
@@ -147,10 +303,46 @@ class ResourceSWAP(qml.SWAP, re.ResourceOperator):
                     0 & 0 & 1 & 0
             \end{bmatrix}.
 
+    .. seealso:: :class:`~.SWAP`
+
     """
 
     @staticmethod
     def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Resources:
+            The resources come from the following identity expressing SWAP as the product of
+            three CNOT gates:
```

Comment:
So the resource decomposition is the one based on qml.SWAP? What's the reference?

Reply:
Just the PL decomposition. This is one of those standard ones you learn in school. I am not sure if there is a reference outside of a textbook maybe. I think the mathematical proof is sufficient reference for any reader.